### PR TITLE
update csharp grammar

### PR DESCRIPTION
**Pull request by cyw3**
update csharp grammar: 
1. change '?' to INTERR; 
2. change '??' to OP_COALESCING; 
3. change method_declaration return type 'VOID' to return_type

### CODE REVIEW

File: csharp/CSharpParser.g4 [hunk 1]
```diff
@@ -691,7 +691,7 @@ common_member_declaration
 	| event_declaration
 	| conversion_operator_declarator (body | right_arrow throwable_expression ';') // C# 6
 	| constructor_declaration
-	| VOID method_declaration
+	| method_declaration  // 2020/10/22 add return_type
```
Comment (KvanTTT) [inline csharp/CSharpParser.g4:694]: Git keeps history, that's why such a comment is excess.

File: csharp/CSharpParser.g4 [hunk 2]
```diff
@@ -1186,8 +1186,9 @@ constructor_declaration
 	: identifier OPEN_PARENS formal_parameter_list? CLOSE_PARENS constructor_initializer? body
 	;
 
+// 2020/10/22 add return_type
```
Comment (KvanTTT) [inline csharp/CSharpParser.g4:1189]: Also, remove this comment.

Comment (cyw3) [reply csharp/CSharpParser.g4]: done

File: csharp/CSharpParser.g4 (hunk 1 above)
Comment (cyw3) [reply csharp/CSharpParser.g4]: done

**Comment by KvanTTT** [comment]
Unfortunately, there are some C# errors on CI: https://github.com/antlr/grammars-v4/pull/1922/checks?check_run_id=1404185801#step:4:2783

**Comment by KvanTTT** [comment]
@cyw3 any chance to fix errors?

**Comment by teverett** [comment]
@cyw3  Thanks for this PR.  It has not been touched since Nov 2020 and will therefore be closed.

**Pull request closed without merging**